crates/ethrpc/src/buffered.rs

```diff
@@ -388,9 +391,11 @@ fn format_indices_as_ranges(indices: BTreeSet<usize>) -> anyhow::Result<String>
         }
     }
 
-    // Append the last range or a single value.
+    // Append the remaining data.
```

Comment:
Just noticed that this code it duplicated.
You could add a function like `append_sequence(buffer, start, end)` which encapsulates it.